### DISCUSSION TITLE
If you are a volunteer or supporter, you can get Pro

### DISCUSSION THREAD

**Discussion by sonvnn** [templaza/astroid-framework#836] [Announcements]
Dear Volunteers & Supporters,

As announced before. I am working on Astroid Pro. It is a plugin of Astroid Framework. It provides more advanced and superior features for Astroid Framework. And I really hope you will join me, as I said in the previous announcement. Those who have contributed and supported the project will have Astroid for free for life, and I promise to keep that promise. So I created this discussion so that you can register to join Astroid Pro repository here.

First of all, I would like to send an invitation to join @Chacapamac  @Joomlaplates  @WM-Loose @paurray @Banuk1616 @iorbita @Dnsftl @salahjaafar

I apologize because I cannot remember all the people who have contributed to the project. So if you have ever participated in testing and contributing even just reporting a few bugs or participating in things that bring Astroid to the Joomla community. Let me know. And if your contributions are worthy, I will bring you on board Astroid Pro.

With my best regards,
Sonny

**Comment by jam1974** [comment]
@sonvnn I have found a few errors and would be happy to be able to search for them in the pro version as well.

**Reply by sonvnn** [reply]
Can you show the detail your contribution to Astroid? 
Thanks,
Sonny

**Reply by jam1974** [reply]
https://github.com/templaza/astroid-framework/issues/807
https://github.com/templaza/astroid-framework/issues/797

and some in discord.

**Reply by sonvnn** [reply]
It would be great, If you help us Translate Astroid into your language, would you please thank you.
Best Regards,
Sonny

**Reply by jam1974** [reply]
Okay, since I grew up bilingual, I can provide the Portuguese language. I have it here.

**Reply by sonvnn** [reply]
@jam1974 Thank you Oliver. Please contact to @Joomlaplates to learn how to create language file for Astroid. I am happy to have you on board :)

**Reply by sonvnn** [reply]
@jam1974 Thank you for helping translate Astroid into Portuguese. Can you update Portuguese language with latest language version? The latest language file is here https://github.com/templaza/astroid-framework/blob/v3/language/en-GB/en-GB.astroid.ini
Thanks & Best Regards,
Sonny

**Reply by jam1974** [reply]
@sonvnn in ASTROID_ENABLE_SHORTCUT_DESC=‘Select if you want to enable keyboard shortcuts for the Astroid backend, available keyboard shortcuts are listed here <a href=’https://docs.joomdev.com/article/hotkeys/‘>https://docs.joomdev.com/article/hotkeys/</a>’ the link refers to an error 404. On https://docs.astroidframe.work/ I can't find anything regarding the keyboard shortcuts. Otherwise, I would have changed the link.

#Update: https://github.com/jam1974/astroid-framework-language-pt/blob/92ae657924d6d88e8ddd879a961cd7fb4d624966/pt-PT.astroid.ini

and 
[astroid_pt-PT.zip](https://github.com/user-attachments/files/17455725/astroid_pt-PT.zip) for @Joomlaplates

**Reply by patrickus33** [reply]
@jam1974  This appears to be an old language file !
joomdev has stopped supporting astroid several years ago!
Si it's normal to have a 404 error page.

**Reply by sonvnn** [reply]
@jam1974 Thank you I will take a time to review all old language line. :(

**Reply by jam1974** [reply]
@patrickus33 I am aware of that. Thank you! I also just wanted to replace the link with an up-to-date source. But I can't find any instructions for the keyboard commands.

**Reply by jam1974** [reply]
@sonvnn Portuguese is online; https://github.com/templaza/astroid-framework/blob/language/astroid_pt-PT.zip

**Comment by patrickus33** [comment]
@sonvnn , my contribution wasn't very important...
Just Version nb error #514 
and this discussion Change after updating template one #774
I also posted a comment on the JED “ [Framework parfait](https://extensions.joomla.org/extension/astroid-framework/#rev-125844)” under the name patrick

**Reply by sonvnn** [reply]
Thanks for helping translate Astroid into French. Invitation sent.

**Comment by web54** [comment]
I'd like to be in there.
For the moment, i just have translated doc in french and had posted some issues, but I can help for some coding as astroid is well coded.
( not a developper, but I understood your code and had already coded some hacks )
thanks for all

**Reply by sonvnn** [reply]
Thank you please help us review and translate Astroid into French. Please send it to @Joomlaplates to update package. I will send you an invite.
Cheers,
Sonny

**Reply by patrickus33** [reply]
@sonvnn 
Hello,
Here is the translation of the file in French.
I tested it on a site and everything is OK.
There will certainly be words or expressions to improve.
The strange thing is that I don’t have translation possibilities for some strings (see screenshot)

[astroid_fr-FR.zip](https://github.com/user-attachments/files/17445379/astroid_fr-FR.zip)


![astroid](https://github.com/user-attachments/assets/b15d1be6-9f64-4155-8c1b-cf9948befb2a)

**Comment by salahjaafar** [comment]
Dear @sonvnn,

I would like to sincerely thank you for the invitation to join Astroid Pro and for acknowledging my contribution to the project. It is truly an honor to be part of this journey and the Astroid community. I am grateful for the opportunity to continue contributing to the development of such an outstanding framework.

I’m excited to explore the new features of Astroid Pro and look forward to collaborating further to enhance this tool for the Joomla community.

Thank you once again for your trust and for this great initiative.

Best regards,
Salah JAAFAR

**Reply by sonvnn** [reply]
Thank you very much for helping me translate Astroid into Arabic and some issue with RTL layout. I appreciate your contribution.

Cheers,
Sonny

**Reply by salahjaafar** [reply]
Continuing this project not only benefits the community but also strengthens our collaboration and helps us all grow together. I look forward to more opportunities to contribute and enhance Astroid further.

**Comment by joomlaplates-de** [comment]
Hello together, I agree 100% with Salah.

**Reply by salahjaafar** [reply]
Thank you for your support! I’m glad we’re on the same page.

Best,
Salah JAAFAR

**Comment by rinka88** [comment]
Thank you for your hard work in developing an excellent template. And to everyone who helps him in his development.

**Comment by patrickus33** [comment]
Hi Sonny,
Is there a place like Transifex where we can update the package ?

Le sam. 19 oct. 2024, 11:08, Sonny Le ***@***.***> a écrit :

> Thank you please help us review and translate Astroid into French. Please
> send it to @Joomlaplates <https://github.com/Joomlaplates> to update
> package. I will send you an invite.
> Cheers,
> Sonny
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/templaza/astroid-framework/discussions/836#discussioncomment-10990053>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6QUZMICXJNXSI4OA72CDDZ4IOPHAVCNFSM6AAAAABQG753YKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOJZGAYDKMY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <templaza/astroid-framework/repo-discussions/836/comments/10990053@
> github.com>
>
Patrick Marangone
0676088910

**Reply by salahjaafar** [reply]
Hi @patrickus33 

You can use a simple text editor like Notepad++ on Windows or Gedit on Linux to update the package. Just make sure to save the file in UTF-8 format to avoid any encoding issues.

Best regards,
Salah JAAFAR

**Reply by patrickus33** [reply]
Hi @salahjaafar , 

I know how to use Notepad++ 😊
Some dev use external platform like Transifex or Crowdin for the translations.
So many users can work together .
That was the meaning of my question

Regards
Patrick

**Reply by joomlaplates-de** [reply]
Hello, I´ve checked the french translation and upload it.
Thanks, merci Peter

**Comment by iorbita** [comment]
Hello Sonny and thank you very much for your invitation and for this great idea. Always ready to help when needed ;)

Many thanks again,
Lorenzo

**Comment by deltapapa01** [comment]
Hey Sonny,
I will be downsizing my business in the next few years but I would be happy to continue to support Astroid and test the Pro features. Even though I don't use the widgets on my sites, it would be a pleasure to continue to be part of the (Pro) team.
Thank you for your effort and commitment,
deltapapa

**Reply by sonvnn** [reply]
Opp! I'm sorry I forgot about you. I sent you an invite because you deserved it. Happy to see you on board :)

**Comment by WM-Loose** [comment]
Hey Sonny,
thank you for the invitation and I'm happy to be part of the project. Together we will make Astroid one of the best frameworks of all time. You can rely on me 100%. By the way, I have already successfully tested Astroid 3.1.6 in Joomla6 Alpha1.
It works very well.

**Comment by Dnsftl** [comment]
@sonvnn,

Thank you very much for the invitation to join Astroid Pro and for acknowledging my contribution to the project. I truly appreciate it and am grateful for the opportunity to contribute to the further development of Astroid.

I also want to sincerely thank you for all the work you’ve already done and the excellent support you’ve provided.

I look forward to continuing to make my contribution to this great project now and in the future. I hope for a pleasant and long-term collaboration.

Once again, thank you for your trust, and I wish Astroid great success in the future.

Best regards,
Dennis

**Comment by Banuk1616** [comment]
Hello,
salahjaafar described it very well. I would agree 100%.
Many thanks and best regards

**Comment by paurray** [comment]
Hello Sonny,

many thanks for the invitation and acknowledgement of work done.
Gratefully accepted.
Am 100% on board and will continue contributing in the way I best can.
Making Training Material.
The more training material I make the more I realise how awesome Astroid is.
I look forward to even more awesomeness all round.

thank you Sonny et al.

Paul

**Reply by sonvnn** [reply]
Because you deserve it. I really like your current work.

**Comment by zenolabs** [comment]
I am available to translate Astroid into italian language.
Best regards.

**Reply by sonvnn** [reply]
Thank you for your interest. Please contribute more to the community at this time. When we see your enthusiasm for our open community. I will immediately send you an invitation. :)

**Comment by WM-Loose** [comment]
The trend today is going in a different direction.

Many developers no longer create language packs and use English as the default.
The browser translation apps are getting better and better and you can then have the template options translated via the browser.

Yes, it is not always 100% correct but it is more than enough to understand it.
Development is progressing so quickly that it is now almost impossible to keep up with the translation and it consumes so many resources.

Instead of creating translations, we should test the beta versions of Astroid in more detail. It is important that all new and existing functions run without problems and that the framework is not only tested with Joomla compatibility, but also with third-party extensions and plugins.

Documentation for handling in different languages ​​might be more useful as translate the template-options?...

**Reply by Chacapamac** [reply]
I think the precision and clarity of a framework's administration language cannot be trusted to an auto-translate system yet. Small errors can be extremely frustrating, and even provoke serious mistakes in the system's management. Maybe one day!

**Reply by jam1974** [reply]
> I think the precision and clarity of a framework's administration language cannot be trusted to an auto-translate system yet. Small errors can be extremely frustrating, and even provoke serious mistakes in the system's management. Maybe one day!

@Chacapamac I can only agree with you - I also prefer to work with the English or Portuguese translations rather than the German - because they tend to be poorly translated and German is my main language.

**Reply by sonvnn** [reply]
@WM-Loose I agree 100% with your point. At the moment we need to pay more attention to testing RC versions to make sure all the features are working properly. Another important thing is that we need more tutorials and videos to improve the user experience so that they understand the product. @paurray  is doing a great job, but we need more help.

**Comment by Brandocracy** [comment]
I added a few suggestions. Thank you for a wonderful framework!

**Comment by felixkat** [comment]
@sonvnn Could you add me to the list for Pro testing?  I'm particularly interested in the "dynamic content" but will happily test all areas.  I'm currently building a website, which will eventually be made public, which tests and demonstrates all features \ widgets.

**Reply by sonvnn** [reply]
@felixkat Thank you for your help in Discussion and Issues. I sent you an invite to Astroid Team. Now you can get Pro and check it.

Best Regards,
Sonny

**Reply by felixkat** [reply]
Thank Sonny, I'll be sending a couple of pull requests over soon, you can see the commits in my fork.

**Comment by darekmer** [comment]
Hello, did I contribute? What do you think? I translated Astroid into Polish, and I can continue doing so.

**Reply by sonvnn** [reply]
I sent you an invitation. Thank you.

**Comment by tmumm** [comment]
@sonvnn not a coder, but made a donation, love to get in on Pro.   Love the project and interested in helping however I can.

**Comment by pacoguio** [comment]
Hello, here is the complete Spanish translation so that you can add it to the project page and the community can benefit. The Spanish translation was a bit outdated. The translation that is now available for download has about 700 lines and the one I am leaving you is updated with 1,700 lines of translation. I hope it helps you.
Thank you very much for your work, I love Astroid, practically all my projects are based on this framework.
Greetings
[astroid_es-ES.zip](https://github.com/user-attachments/files/17823206/astroid_es-ES.zip)

**Reply by joomlaplates-de** [reply]
Hola, muchas gracias.

**Reply by pacoguio** [reply]
You're welcome. I don't know if this will serve as a contribution to let me use the pro version of Astroid. If so, I would be very grateful. In any case, you can count on me to update the translation of the framework into Spanish or other things in the future if they are within my possibilities.
Greetings

**Reply by schabert68** [reply]
Could you add me for the list? I interested in the "dynamic content"

**Reply by sonvnn** [reply]
@pacoguio Thank you for your contribution. I have sent you an invitation to join the Astroid Team. Please be active and support us in growing the Astroid community.

**Reply by sonvnn** [reply]
@schabert68 Show that you are an active contributor and supporter of the community. We will be happy to send you an invite, if you are worthy.

**Reply by pacoguio** [reply]
Ok, but I already said that I can help you with translations, I'm not a programmer, I know how to do some small development things and design things. It's not much, but it's what I know.

**Reply by sonvnn** [reply]
@pacoguio So you simply need to follow us. Since the Framework is constantly evolving, I need you to help update your language as soon as a new version is released.

**Reply by pacoguio** [reply]
Ok, then take it for granted. As soon as new strings come out that need to be translated, let me know, send me the file with what needs to be translated and I'll get on with it. I have already collaborated on other occasions with projects such as Acymailing, Seblod and Joomla itself.

**Reply by pacoguio** [reply]
Hello, I have been sent a revision of the Spanish language file. I have already reviewed it, made some improvements and I think everything is correct. I don't know where I should send the corrected file. For now I leave it here so that you can put it in the download area if necessary. Thanks
[astroid_es-ES.zip](https://github.com/user-attachments/files/17879252/astroid_es-ES.zip)

**Reply by pacoguio** [reply]
Hi, the file is already checked. The language strings that were missing in the cache module have already been translated. Here is the file. Regards
[astroid_es-ES.zip](https://github.com/user-attachments/files/17899206/astroid_es-ES.zip)

**Reply by pacoguio** [reply]
@sonvnn I don't know if I'm doing the right thing by leaving the files here. I'm leaving the revised and translated files in this thread, but I suppose you have a more specific place to send the finished files other than here in this thread. If so, please give me instructions. Thanks

**Reply by sonvnn** [reply]
@pacoguio There are 2 ways to update the language.
1. You fork Astroid to your git. Update the language pack and then create a pull request to Astroid. I will help you merge it to the language branch.
2. The second way is you submit the language pack and ask @Joomlaplate to help to update it to the language branch. He is very friendly and will help you do that.

**Reply by pacoguio** [reply]
The previous revision was for the general language file, here is the individual translation of the cache module
[es-ES_mod_astroid_clear_cache.zip](https://github.com/user-attachments/files/17899585/es-ES_mod_astroid_clear_cache.zip)

**Reply by sonvnn** [reply]
Thank you I will update it to the git.
